nx/lib/nx/binary_backend.ex

```diff
@@ -1572,17 +1573,32 @@ defmodule Nx.BinaryBackend do
   def bitcast(out, tensor), do: from_binary(out, to_binary(tensor))
 
   @impl true
-  def sort(_out, t, opts) do
+  def sort(output, t, opts), do: do_sort(output, t, opts, false)
+
+  @impl true
+  def argsort(output, t, opts), do: do_sort(output, t, opts, true)
+
+  defp do_sort(output, t, opts, return_indices) do
     %T{shape: shape, type: type} = t
     last_axis = Nx.rank(t) - 1
 
+    axis = opts[:axis]
+
     comparator =
       case opts[:comparator] do
         :desc ->
-          &</2
+          fn a, b ->
+            a = binary_to_number(a, type)
+            b = binary_to_number(b, type)
+            a <= b
+          end
 
         :asc ->
-          &>/2
+          fn a, b ->
+            a = binary_to_number(a, type)
+            b = binary_to_number(b, type)
+            a >= b
+          end
```

Comment:
These changes are two-fold. First because the original comparators didn't work with negative numbers, and secondly I've used`<=`  and `>=` so the sort is stable